kyo-prelude/shared/src/main/scala/kyo/Env.scala

```diff
@@ -29,10 +29,12 @@ object Env:
     ): A < (S & reduce.SReduced) =
         reduce(ContextEffect.handle(erasedTag[R], env, _.union(env))(v): A < (Env[VR] & S))
 
-    transparent inline def runLayer[A, S, V](inline layers: Layer[?, ?]*)(value: A < (Env[V] & S)): A < (S & Memo) =
-        Layer.init[V](layers*).run.map { env =>
-            runTypeMap(env)(value)
-        }.asInstanceOf[A < (S & Memo)]
+    transparent inline def runLayer[A, S, V](inline layers: Layer[?, ?]*)(value: A < (Env[V] & S)): A < (S & Memo & Nothing) =
```

Comment:
Is the `Nothing` required?

Reply:
I imagine there's some limitation with the macro but wouldn't `runLayer[A, S1, S2, V](inline layers: Layer[?, S1]*)(value: A < (Env[V] & S2))` work?

Reply:
Doesn't work without including `Nothing`: `A < (S & Memo & Nothing)` provides an upper bound that the `transparent inline` can make more specific.

I also wasn't able to make it work with the type parameter `S1` in `Layer[?, S1]`. I'm not really sure I understand why, however. It may be that it lifts all the `In` parameters in `Layer[Out, Int]` to the same type, which makes the macro not work properly.